edx_rest_api_client/tests/test_client.py

```diff
@@ -21,20 +23,29 @@
 class EdxRestApiClientTests(TestCase):
     """ Tests for the E-Commerce API client. """
 
+    @ddt.unpack
     @ddt.data(
-        {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': FULL_NAME, 'email': EMAIL},
-        {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': None, 'email': EMAIL},
-        {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': FULL_NAME, 'email': None},
-        {'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': None, 'email': None},
+        ({'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME,
+          'full_name': FULL_NAME, 'email': EMAIL}, JwtAuth),
+        ({'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': None, 'email': EMAIL}, JwtAuth),
+        ({'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME,
+          'full_name': FULL_NAME, 'email': None}, JwtAuth),
+        ({'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME, 'full_name': None, 'email': None}, JwtAuth),
+        ({'url': URL, 'signing_key': SIGNING_KEY, 'username': USERNAME}, JwtAuth),
+        ({'url': URL, 'signing_key': None, 'username': USERNAME}, NoneType),
+        ({'url': URL, 'signing_key': SIGNING_KEY, 'username': None}, NoneType),
+        ({'url': URL, 'signing_key': None, 'username': None, 'oauth_access_token': None}, NoneType)
     )
-    def test_valid_configuration(self, kwargs):
-        """ The constructor should return successfully if all arguments are valid. """
-        EdxRestApiClient(**kwargs)
+    def test_valid_configuration(self, kwargs, auth_type=None):
```

Comment:
nitpick: the default value for `auth_type` isn't needed, since it's explicitly specified with ddt.data.